control_panel_api/views.py

```diff
@@ -48,3 +49,14 @@ class S3BucketViewSet(viewsets.ModelViewSet):
     serializer_class = S3BucketSerializer
     filter_backends = (S3BucketFilter,)
     permission_classes = (S3BucketPermissions, )
+
+    def perform_create(self, serializer):
+        instance = serializer.save()
+        services.create_bucket(instance.name)
+        services.create_bucket_policies(instance.name)
```

Comment:
I think being explicit is better than implicit.  If it was coupled inside create_bucket then we would have to do the same with the delete_bucket - but as just demonstrated we don't want to delete the bucket at the same time as the policy.

Reply:
I don't know, I thought the idea behind having the `services` module was that it would orchestrate these things for the client code. e.g. you ask `services` "create an S3 bucket" and it would take care of it.

It's similar to ask a waiter a dish, the waiter just knows who to ask the dish, not how to cook it, what kind of ingredients needs,  etc...I see `services` as an interface between our app code and AWS (in this case) => the app wants an S3 bucket, what happens behind the scenes is not important from the view perspective.